### PR TITLE
Dvorak submaps!

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds submaps (randomization of some ruin rooms) to the telecomnuication sat
Adds some spawners to go with it.
Adds ambience.

<details>
<summary>Want to ruin the surprise? I don't recommend it.</summary>
<br>
A new secbot that explodes, shows up in one submap.
image
5 turret rooms
image
4 vending machine rooms.
Spawner with a 1/20 to spawn the doomba
spawner with a 1/25 to spawn a welding tank
placed said spawner under a certain door
Made the shield projector start under a turret cover, get that emp
</details>

## Why It's Good For The Game

Submaps are cool. Let's use them

## Testing
Spawning in the various submaps

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
add: Dvorak's ruin has some variations.
add: Ambience to dvorak's ruin.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Just so you know, the title made me think you were adding sub keymaps for the DVORAK keyboard layout.

**Comment by dj-34** [comment]
is it neccessary to make such large maps for spawners?

**Comment by Qwertytoforty** [comment]
> is it neccessary to make such large maps for spawners?

No, I could probably shrink them down. I wasn't sure how many I would make so I put in 80 by 80 initally

**Comment by warriorstar-orion** [comment]
> > is it neccessary to make such large maps for spawners?
> 
> No, I could probably shrink them down. I wasn't sure how many I would make so I put in 80 by 80 initally

I just started putting all mine in one file. Still feeling out the best way to handle these but this is pretty convenient and I don't have to jump back and forth between files.

image

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 1]
```diff
@@ -17,6 +17,36 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
+	desc = "IT'S GOT A BOMB RUN!"
+	var/obj/structure/reagent_dispensers/fueltank/internal_tank
+	var/obj/structure/marker_beacon/dock_marker/collision/decorative_eye
+
+/mob/living/simple_animal/bot/secbot/buzzsky/telecomms/doomba/Initialize(mapload)
+	. = ..()
+	internal_tank = new /obj/structure/reagent_dispensers/fueltank(src)
+	internal_tank.forceMove(src)
+	decorative_eye = new /obj/structure/marker_beacon/dock_marker/collision(src)
+	decorative_eye.forceMove(src)
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:31]: Also shouldn't be needed

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 2]
```diff
@@ -247,6 +277,26 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
+	invisibility = 90
+	var/obj/machinery/shieldgen/telecomms/trap = new /obj/machinery/shieldgen/telecomms(get_turf(src))
+	var/atom/flick_holder = new /atom/movable/porta_turret_cover(loc)
+	flick_holder.layer = trap.layer + 0.1
+	flick("popup", flick_holder)
+	sleep(1 SECONDS)
+	trap.shields_up()
+	qdel(flick_holder)
+	qdel(src)
+
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:298]:
```suggestion
```

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 3]
```diff
@@ -247,6 +277,26 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
+	icon_state = "turretCover"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	anchored = TRUE
+
+/obj/structure/telecomms_shield_cover/proc/activate()
+	invisibility = 90
+	var/obj/machinery/shieldgen/telecomms/trap = new /obj/machinery/shieldgen/telecomms(get_turf(src))
+	var/atom/flick_holder = new /atom/movable/porta_turret_cover(loc)
+	flick_holder.layer = trap.layer + 0.1
+	flick("popup", flick_holder)
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:293]: Didn't flick have issues sometimes?

Comment (Qwertytoforty) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:293]: It does related to how often flick is played. Ideally it was fully purged, yes, but something that flicks once is much less of a problem.

File: _maps/map_files/RandomRuins/SpaceRuins/telecomns_returns.jsonc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        "submap_size_x": 9,
+        "submap_size_y": 5,
+		"submap_size_z": 1,
+        "submaps_dmm": "submaps/DVORAK_turret_room.dmm",
+        "marker_extract": "/obj/effect/map_effect/marker/mapmanip/submap/extract/space_ruin/dvorak/turret_room",
+        "marker_insert": "/obj/effect/map_effect/marker/mapmanip/submap/insert/space_ruin/dvorak/turret_room",
+        "submaps_can_repeat": true // doesn't matter, as there's only one insert marker
+    }
+]
+
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply _maps/map_files/RandomRuins/SpaceRuins/telecomns_returns.jsonc:25]: tabbed

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm (hunk 1 above)
Comment (Qwertytoforty) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:31]: yeah it won't, I was gaslighting myself by not making them click opaque originally which made them appear on right click

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm (hunk 2 above)
Comment (Qwertytoforty) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:298]: removed

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm (hunk 3 above)
Comment (DGamerL) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:293]: Sounds good then

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 5]
```diff
@@ -17,6 +17,34 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
 
+/mob/living/simple_animal/bot/secbot/buzzsky/telecomms/doomba
+	name = "A FUCKING DOOMBA"
+	desc = "IT'S GOT A BOMB RUN!"
+	var/obj/structure/reagent_dispensers/fueltank/internal_tank
+	var/obj/structure/marker_beacon/dock_marker/collision/decorative_eye
+
+/mob/living/simple_animal/bot/secbot/buzzsky/telecomms/doomba/Initialize(mapload)
+	. = ..()
+	internal_tank = new /obj/structure/reagent_dispensers/fueltank(src)
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:28]:
```suggestion
	internal_tank = new(src)
```

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 6]
```diff
@@ -17,6 +17,34 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
+	decorative_eye = new /obj/structure/marker_beacon/dock_marker/collision(src)
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:29]:
```suggestion
	decorative_eye = new(src)
```

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm [hunk 7]
```diff
@@ -391,6 +438,18 @@ GLOBAL_LIST_EMPTY(telecomms_trap_tank)
...
+	result = list(
+		/datum/nothing = 3,
+		/obj/structure/reagent_dispensers/fueltank = 1)
+
+/obj/effect/spawner/random_spawners/telecomms_doomba_one_in_twenty
+	name = "weldertank maybe"
+	result = list(
+		/datum/nothing = 19,
+		/mob/living/simple_animal/bot/secbot/buzzsky/telecomms/doomba = 1)
+
```
Comment (DGamerL) [inline code/modules/awaymissions/mission_code/ruins/telecomns.dm:452]: image

**Review by DGamerL — CHANGES_REQUESTED** [review]
Just these last bits and I'm fine with the DM side

File: code/modules/awaymissions/mission_code/ruins/telecomns.dm (hunk 7 above)
Comment (Qwertytoforty) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:452]: hehehehe, no-
Oh wait I see what he wants. I read this as spawn 25 welder tanks when I first read it lmao

Comment (DGamerL) [reply code/modules/awaymissions/mission_code/ruins/telecomns.dm:452]: Lmfao, happens

**Review by DGamerL — APPROVED** [review]

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]
Could you flip the submap edges around so that they're placed on the outside of the templates?

2024_11_01__02_22_14__paradise dme  DVORAK_submap dmm  - StrongDMM

I also think the naming scheme should be original_map_name.submaps.dmm for files that have multiple submaps. (if they were different files it would probably be telecomns_returns.vendor_room.dmm etc.)

**Review by warriorstar-orion — CHANGES_REQUESTED** [review]
Also, if all these elements are the same across the submaps I'd keep them in the main map and leave them out of the submaps.

image

**Comment by Qwertytoforty** [comment]
> Also, if all these elements are the same across the submaps I'd keep them in the main map and leave them out of the submaps.
> 
> image

Lights and holopad I included if people wanted to move it, or in the case of the xeno equivalent in vending, remove lights. Button and rust should be consistant though, you are right. Will rename the map file.

**Review by warriorstar-orion — APPROVED** [review]

File: code/game/machinery/camera/camera_presets.dm [hunk 8]
```diff
@@ -33,6 +33,84 @@
...
+		if(SOUTH)
+			camera_overlay.pixel_x = 4
+			camera_overlay.pixel_y = 20
+
+		if(WEST)
+			camera_overlay.pixel_x = 20
+			camera_overlay.pixel_y = 23
+
+	camera_overlay.dir = dir
+	update_icon(UPDATE_OVERLAYS)
```
Comment (DGamerL) [inline code/game/machinery/camera/camera_presets.dm:63]:
```suggestion
```
I forgot that we don't need this anymore

File: code/game/machinery/camera/camera_presets.dm [hunk 9]
```diff
@@ -33,6 +33,84 @@
...
+/obj/machinery/camera/tracking_head/update_overlays()
+	return
+
+/obj/machinery/camera/tracking_head/HasProximity(atom/movable/AM)
+	if(!isrobot(AM) && !iscarbon(AM)) // Only care about carbons and borgs.
+		return
+	if(camera_overlay.dir == get_dir(src, AM))
+		return
+	camera_overlay.dir = get_dir(src, AM)
+	update_icon(UPDATE_OVERLAYS)
```
Comment (DGamerL) [inline code/game/machinery/camera/camera_presets.dm:86]:
```suggestion
```
We also don't need this one anymore I think

**Review by DGamerL — APPROVED** [review]
Wow what beautiful camera code

**Review by DGamerL — APPROVED** [review]

**Pull request merged**